src/components/app/SingleStrategy/index.tsx

```diff
@@ -98,7 +118,7 @@ export const SingleStrategy = () => {
                         details={error}
                     />
                 )}
-                {isLoading || isReportsLoading ? (
+                {isLoading || isVaultLoading || isReportsLoading ? (
```

Comment:
how good is the performance of waiting to load all three? another approach is do a label just for the name like
```
 <StyledSpan>
            (Vault:  { isVaulLoading? '...loading vault name': vault.name})
```

Reply:
That is a very good approach. On localhost I could see the loading text, so it does help loading time.